Recruit-Training/Advanced-Recruit-Training/Telemetry-Teaser/src/App.tsx

```diff
@@ -2,24 +2,66 @@ import BatteryInput from "~/components/batteryInput";
 import Header from "~/components/header";
 import SpeedInput from "~/components/speedInput";
 import WeatherInput from "~/components/weatherInput";
+import Button from "~/components/Button/Button.tsx";
+import {useState} from "react";
 
 const App = () => {
-  const calculateRange = () => {
-    return;
+  const [speed,setSpeed] = useState(null);
+  const [batteryPercent, changeBatteryPercent] = useState(0);
+  const [weather, weatherChange] = useState(0);
+  const[speedValid,setSpeedValid]= useState(false);
+  const[batteryValid,setBatteryValid]= useState(false);
+  const [calc, setCalc] = useState(0)
+  const [initialCalc, setInitialCalc] =useState(false);
```

Comment:
Do we need to create extra state to store the calculated range? We should be able to derive the range from the 3 state variables `speed`, `batteryPercent`, `weather`.

Remember, when you update state in React, it will cause your whole component to re-render (unless you use useMemo or useCallback...will get to that in future lesson). Think about using this to your advantage and you may not need extra state to store the range.